examples/fluids/src/misc.c

```diff
@@ -174,8 +175,10 @@ PetscErrorCode PostProcess_NS(TS ts, CeedData ceed_data, DM dm, ProblemData *pro
 
   // Print final time and number of steps
   PetscCall(TSGetStepNumber(ts, &steps));
+  PetscCall(TSGetConvergedReason(ts, &reason));
   if (!user->app_ctx->test_mode) {
-    PetscCall(PetscPrintf(PETSC_COMM_WORLD, "Time integrator took %" PetscInt_FMT " time steps to reach final time %g\n", steps, (double)final_time));
+    PetscCall(PetscPrintf(PETSC_COMM_WORLD, "Time integrator %s on time step %" PetscInt_FMT " with final time %g\n", TSConvergedReasons[reason],
```

Comment:
@jrwrigh @jeremylt I think this is more accurate in case of restart. Should we explicitly state the starting time step in this output?

Reply:
> Should we explicitly state the starting time step in this output?

I don't think it's necessary per-say since the log itself will effectively say which timestep it started on. But I usually lean towards redundancy for log output as long as it isn't overwelming. And I don't think adding a `"Ran from time steps ____ to ____"` makes it overwhelming. Though it's make sense to do that with the solution time as well.

Reply:
If you run with no monitors, you get something like this
```console
$ mpiexec -n 5 build/fluids-navierstokes -options_file examples/fluids/newtonianwave.yaml

-- Navier-Stokes solver - libCEED + PETSc --
  MPI:
    Host Name                          : kichatna
    Total ranks                        : 5
  Problem:
    Problem Name                       : newtonian_wave
    Stabilization                      : SUPG
  libCEED:
    libCEED Backend                    : /cpu/self/xsmm/blocked
    libCEED Backend MemType            : host
  PETSc:
    Box Faces                          : 40,40,1
    DM MatType                         : aij
    DM VecType                         : standard
    Time Stepping Scheme               : implicit
  Mesh:
    Number of 1D Basis Nodes (P)       : 2
    Number of 1D Quadrature Points (Q) : 2
    Global DoFs                        : 13448
    Owned DoFs                         : 2616
    DoFs per node                      : 5
    Global nodes (DoFs / 5)            : 2689
    Local nodes                        : 736
Time taken for solution (sec): 2.22668
Time integrator CONVERGED_ITS on time step 100 with final time 0.2
```

Perhaps we should announce the starting step/time and intended final step/time before calling `TSSolve`. More wording suggestions welcome.

Reply:
Something at the beginning like: 
```
Restart file found: <path to file>
Loading initial condition from restart at step <step num> and solution time <solution time>
```

I don't think the intended final step/time is necessary. It would actually be nice to be able to change that on the fly, but that's a convenience feature for a (probably much) later time.

Reply:
Change it by writing to a control file or socket?

Reply:
> Change it by writing to a control file or socket?

PHASTA's implementation uses a control file to change the ending timestep. At the beginning of every timestep, it looks for a specific file. If it's present, it reads it, changes the end timestep, and deletes the file. Otherwise it just keeps solving.

Reply:
Revised to print this
```
  Continue:
    Filename:                          : ./ns-solution.bin
    Step:                              : 10
    Time:                              : 0.02
```